crates/paralegal-policy/src/context.rs

```diff
@@ -462,6 +453,15 @@ impl Context {
             let Some(datasource) = current.typ.as_data_source() else {
```

Comment:
I know this isn't quite the point of this PR, but I would recommend `queue` should always contain data sources + controller id's (to get the flow). Instead I would convert the inputs to data sources once, only use data sources in the queue and create the nodes just inline to call the `is_terminal` and `is_checkpoint` functions with.
On one hand that means you can avoid this inline "error handling", but also if you do it at the beginning you can use `assert_error!` or `assert_warning!` to let the user know that they provided an invalid starting point.

Reply:
Hm, I think it's valid for the user to provide a starting point that isn't a DataSource, though. I believe the only node that can't be converted using as_data_source is the controller Return. For example, if their starting points are all nodes marked `sensitive`, and there are some sensitive nodes that are arguments or callsites or whatever but the controller Return is also marked sensitive, I don't think that always_happens_before should just fail.

Reply:
Yes of course it's valid to provide a non-DataSource argument. But we can convert that before the loop.

In terms of the sensitive `Return`, I would be agreeable to warning about the failed conversion instead of a hard error.

Reply:
Updated this function. Can you take another look and see if it makes sense?